README.md

```diff
@@ -159,6 +159,41 @@ The roadmap and all features of this plugin are open to change, and any suggesti
   * List all definitions
   * Go to definition
 
+## Defining Modules
+
+Users and plugin authors can take advantage of modules by creating their own. Modules provide:
+
+- Treesitter language detection support
+- Attach and detach to buffers
+- Works with all nvim-treesitter commands
+
+You can use the `define_modules` function to define one or more modules or module groups.
+
+```lua
+require 'nvim-treesitter'.define_modules {
+  my_cool_plugin = {
+    attach = function(bufnr, lang)
+      -- Do cool stuff here
+    end,
+    detach = function(bufnr)
+      -- Undo cool stuff here
+    end,
+    is_supported = function(lang)
+      -- Check if the language is supported
+    end
+  }
+}
+```
+
+Modules can consist of the following properties:
+
+- `module_path`: A require path (string) that exports a module with an `attach` and `detach` function. This is not required if the functions are on this definition.
+- `enable?`: Determines if the module is enabled by default.
+- `disable?`: A list of languages that this module is disabled for. This is usually overridden by the user.
+- `is_supported?`: A function that takes a language and determines if this module supports that language.
+- `attach?`: A function that attachs to a buffer. This is required if `module_path` is not provided.
+- `detach?`: A function that detaches from a buffer. This is required if `module_path` is not provided.
```

Comment:
Why adding `?` here ? Because they are optional ?

Reply:
Yeah... Sorry. I write a lot of typescript and that's the notation for optional. What do you prefer to mark them as optional.

Reply:
Hmmm, maybe just add a notice before all of that, the notation is fine once this is explained.